.github/workflows/pytest.yaml

```diff
@@ -6,22 +6,17 @@ jobs:
   pytest:
     name: Run tests
     runs-on: ubuntu-latest
+    container:
+      image: python:3.12-bullseye
 
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
 
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.12'
-          cache: 'pip'
```

Comment:
why is this removed? I understand that Python is pre-installed in the image, but this also takes care of caching the installed dependencies.

Reply:
With the build process now running inside a container, I don't think this type of caching still works. I did run into some weird issues with the setup-python action combined with the Docker container so I decided to remove it. Maybe someone with more knowledge about Github CI has an explanation for that.

Reply:
Whatever is the issue there, removing the caching here is a net loss, so I'm not a fan of this change. If `setup-python` is not working, then please set up https://github.com/actions/cache as a replacement instead, thanks.

Reply:
I have reverted the changes to pytest.yaml, now it's just the release.yml that has changed. Since a release isn't happening that often, do we still need working caching for that?

Reply:
the release workflow can stay without cache.